test/parallel/test-vm-property-not-on-sandbox.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+require('../common');
+const assert = require('assert');
+const vm = require('vm');
+
+// These, admittedly contrived, examples test two edge cases of the vm module.
+//
+// Both GetterCallback and QueryCallback
+// explicitly check the global_proxy() if a property is
+// not found on the sandbox. In the following tests, the explicit check
+// inside the callback yields different results than deferring the
+// check until after the callback. The check is deferred, if the
+// callbacks do not intercept, i.e., if args.GetReturnValue().Set() is
+// not called.
+
+
+// Check that the SetterCallback explicitly calls GetRealNamedProperty()
+// on the global proxy if the property is not found on the sandbox.
+//
+// foo is not defined on the sandbox until we call CopyProperties().
+// In the GetterCallback, we do not find the property on the sandbox and
+// get the property from the global proxy. Since the return value is
+// the sandbox, we replace it by
+// the global_proxy to keep the correct identities.
+let sandbox = {console};
+sandbox.document = { defaultView: sandbox };
+vm.createContext(sandbox);
+const code1 = 'Object.defineProperty(' +
```

Comment:
We've been moving toward using block scopes in tests to avoid things like `code1`, `code2`, etc.